pghoard/config.py

```diff
@@ -159,6 +160,124 @@ def read_json_config_file(filename, *, check_commands=True, add_defaults=True, c
     return set_and_check_config_defaults(config, check_commands=check_commands, check_pgdata=check_pgdata)
 
 
+def validate_config_file(config_fp):
+    config = json.load(config_fp)
+
+    NUMERIC_RE = re.compile(r"\d+")
+
+    def is_boolean(b):
+        return b in ("true", "false", "True", "False")
+
+    def str_not_blank(s):
+        return s and len(s.strip())
+
+    def validate(rootdict, field, validator, has_default=True):
+        field_val = str(rootdict.get(field, ""))
```

Comment:
You shouldn't cast these all to str since like mentioned in the boolean case, we actually rely in the code that it is actual python boolean.